lib/swoosh/adapters/amazon_ses.ex

```diff
@@ -0,0 +1,203 @@
+defmodule Swoosh.Adapters.AmazonSes do
+  @moduledoc ~S"""
+  An adapter that sends email using the Amazon Simple Email Service Query API.
+
+  For reference: [Amazone SES Query Api Docs](http://docs.aws.amazon.com/ses/latest/APIReference/Welcome.html)
+
+  ## Example
+
+      # config/config.exs
+      config :sample, Sample.Mailer,
+        adapter: Swoosh.Adapters.AmazonSes,
+        region: "region-endpoint",
+        access_key: "aws-access-key",
+        secret: "aws-secret-key"
+
+      # lib/sample/mailer.ex
+      defmodule Sample.Mailer do
+        use Swoosh.Mailer, otp_app: :sample
+      end
+  """
+
+  use Swoosh.Adapter, required_config: [:region, :access_key, :secret]
+  alias Swoosh.Email
+  alias Swoosh.Adapters.XML.Helpers, as: XMLHelper
+  alias Swoosh.Adapters.SMTP.Helpers, as: SMTPHelper
+
+  @encoding     "AWS4-HMAC-SHA256"
+  @host_prefix  "email."
+  @host_suffix  ".amazonaws.com"
+  @service_name "ses"
+  @action       "SendRawEmail"
+  @base_headers %{"Content-Type" => "application/x-www-form-urlencoded"}
+  @version      "2010-12-01"
+
+  def deliver(%Email{} = email, config \\ []) do
+    query = email |> prepare_body(config) |> encode_body
+    url = base_url(config)
+    headers = prepare_headers(@base_headers, query, config)
+
+    case :hackney.post(url, headers, query, [:with_body]) do
+      {:ok, 200, _headers, body} ->
+        {:ok, interpret_response(body)}
+      {:ok, code, _headers, body} when code > 399 ->
+        {:error, interpret_error_response(body)}
+      {_, reason} ->
+        {:error, reason}
+    end
+  end
+
+  defp interpret_response(body) do
+    id =
+      body
+      |> XMLHelper.parse
+      |> XMLHelper.first("//MessageId")
+      |> XMLHelper.text
+
+    %{id: id}
```

Comment:
Is there anything else useful in that response body? If there is, feel free to add it to the result map.